torch/_export/passes/add_runtime_assertions_for_constraints_pass.py

```diff
@@ -194,4 +158,58 @@ def placeholder(self, name: str, arg, meta) -> ProxyValue:
         # TODO Add relational constraints
         self.input_tracker += 1
         return arg
-    # TODO implement adding inline constraints as assertion
+
+
+    def _assert_constraint(self, proxy, lower, upper, assert_msg, low_threshold=2):
+        if lower > low_threshold:
+            self._inser_assert_async(operator.ge, proxy, lower, assert_msg)
+
+        if upper < math.inf:
+            self._inser_assert_async(operator.le, proxy, upper, assert_msg)
+
+    def _inser_assert_async(self, operator, l, r, assert_msg):
+        cmp = super().call_operator(operator, (l, r), {}, NodeMetadata({}))
+        cmp_tensor = super().call_operator(torch.ops.aten.scalar_tensor.default, (cmp,), {}, NodeMetadata({}))
+        super().call_operator(
+            torch.ops.aten._assert_async.msg,
+            (cmp_tensor, assert_msg),
+            {},
+            NodeMetadata({}),
+        )
+
+    def call_operator(self, op, args, kwargs, meta) -> ProxyValue:
+        ret = super().call_operator(op, args, kwargs, meta)
+        if "val" in meta:
+            val = meta["val"]
+
+            def add_assertions(val):
+                call_backs = []
+                messages = []
+                if isinstance(val, (torch.SymInt, torch.SymFloat, torch.SymBool)):
+                    expr = val.node._expr
+                    if expr in self.inline_constraints:
+                        constraint = self.inline_constraints[expr]
+                        lower = _convert_to_int(constraint.lower)
+                        upper = _convert_to_int(constraint.upper)
+                        assert_msg = f" is outside of inline constraint [{lower}, {upper}]."
+                        call_backs.append(partial(self._assert_constraint, lower=lower, upper=upper, low_threshold=-1))
+                        messages.append(assert_msg)
+                elif isinstance(val, torch.Tensor):
+                    for i, sym in enumerate(val.shape):
+                        cbs, msgs = add_assertions(sym)
+                        for cb, msg in zip(cbs, msgs):
+                            def sym_size_cb(proxy, assert_msg, dim):
+                                dim_proxy = super(AddRuntimeAssertionsForConstraintsPass, self).call_operator(
```

Comment:
Why can't it be just super().call_operator()?

Reply:
I'm not sure.. Seems related to some weird  name resolution issue.